ports/espressif/common-hal/wifi/Radio.c

```diff
@@ -250,6 +250,47 @@ void common_hal_wifi_radio_stop_ap(wifi_radio_obj_t *self) {
     set_mode_ap(self, false);
 }
 
+mp_obj_t common_hal_wifi_radio_get_stations_ap(wifi_radio_obj_t *self) {
+    wifi_sta_list_t esp_sta_list;
+    esp_netif_pair_mac_ip_t mac_ip_pair[ESP_WIFI_MAX_CONN_NUM];
```

Comment:
Thanks for diving into this, it will be quite handy.

I tried to track it down, but I'm not sure where `ESP_WIFI_MAX_CONN_NUM` comes from. From your tests, it appears to be 15 for some Espressif boards, 10 for others. I thought 10 used to be the max, but seems to be 15 now. However, the following seems to indicate that 10 is esp-idf default (and it trades off with number of ESP-NOW encrypted pairs):
https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32s3/api-guides/wifi.html#ap-basic-configuration

Also not sure what code params you used for `wifi.radio.start_ap()`, but default max_connections has historically always been 4. Should `get_stations_ap()` and `get_max_stations_ap()` be capped at the `max_connections` param in `start_ap()`? There is limit checking for `max-connections` with a max of 10, so if it's actually 15, that should change.

Reply:
(fwiw, ESP-NOW maximum number of encrypted peers is a build-time option, whereas maximum number of connections of SoftAP is a runtime option - [link](https://github.com/espressif/esp-idf/issues/11697))

Reply:
I appreciate ESP-IDF's move to extremely modular components but sometimes things are scattered to the winds :)

I'd thought of `get_max_stations_ap()` as the upper limit of stations the firmware was configured to support, but I see the argument that it shouldn't report more than `max_connections`. I'd missed the part about sharing keys with ESP-NOW, good catch.

I don't think anybody asked for `get_max_stations_ap()`. I included it because it was easy and I thought it might help one or two users to be able to query how many stations the device supports. I doubt anyone's code is going to need it. So I don't have strong feelings about including it - if it complicates things or may be misleading, maybe it's better to leave it out. If you'd like to keep it, I'm happy to change it to `max_connections` <= 17 - `CONFIG_ESP_WIFI_ESPNOW_MAX_ENCRYPT_NUM` though I really wish Espressif had provided a #defined constant to use rather than hard coding `17`.

Reply:
Agreed on the `17`.

I can't actually find the `7` (ESP-Now) anywhere, perhaps it's in the esp-idf component and we just don't have it defined differently in our config files. I've never tested more than 2 encrypted pairs.

I don't have a strong opinion about `get_max_stations_ap()` either. Since there is an out of bounds exception in the `espressif` port for `max-connections`, maybe it's superfluous. `raspberry` afaik is fixed at 10.

Reply:
I've removed `get_max_stations_ap()`. Might as well save the bytes and potential confusion until someone actually asks for it. It's probably better suited to documentation anyway. Thanks for catching the `max_connections` stuff!

I'm still using `ESP_WIFI_MAX_CONN_NUM` internally as an upper bound for temporary storage while querying about connected stations, but that should be a safe use.